File: 338_Counting_Bits/338_2.c
int* countBits(int num, int* returnSize){
    *returnSize = num + 1;
    int n = *returnSize;
    int *res = (int *)malloc(sizeof(int) * n);
    for (int i = 0; i < n; i++) {
        res[i] = 0;
    }
    for (int i = 1; i <= num; i++) {
        res[i] = res[i & (i - 1)] + 1;
    }
    return res;
}

File: 338_Counting_Bits/338_2.py
class Solution:
    def countBits(self, num: int) -> List[int]:
        res = [0] * (num + 1)
        for n in range(1, num + 1):
            count = 0
            m = n
            while m > 0:
                m &= (m - 1)
                count += 1
            res[n] = count
        return res
